docs/metrics/bleu.md

```diff
@@ -0,0 +1,132 @@
+
+    **Reference Sentence**:
+
+    | n | n-Grams |
+    | ---   | ---                        |
+    | 1 | ["kolena", "is", "a", "comprehensive", "machine", "learning", "testing", "and", "debugging", "platform", "to", "surface", "hidden", "model", "behaviors", "and", "take", "the", "mystery", "out"]|
+    |...|
+    | 4 | ["kolena is a comprehensive", "is a comprehensive machine", "a comprehensive machine learning", "comprehensive machine learning testing", "machine learning testing and", "learning testing and debugging", "testing and debugging platform", "and debugging platform to", "debugging platform to surface", "platform to surface hidden", "to surface hidden model", "surface hidden model behaviors", "hidden model behaviors and", "model behaviors and take", "behaviors and take the", "and take the mystery", "take the mystery out", "the mystery out of", "mystery out of model", "out of model development"]  |
+
+??? example "Step 2: Calculate n-Gram Overlap"
+    Next, lets calculate the clipped precision scores for each of the n-Grams. Recall that the precision formula is:
+
+    $$
+    p_i = \frac{\text{Clipped} \text{ count of matching i-grams in machine-generated text}^1}{\text{Total number of i-grams in machine-generated text}}
+    $$
+
+    <center>
+
+    | n | Clipped Precision  |
+    | --- | --- |
+    | 1 | 14 / 17 = 0.824 |
+    | 2 | 10 / 16 = 0.625 |
+    | 3 | 6 / 15 = 0.400 |
+    | 4 | 3 / 14 = 0.214 |
+
+    </center>
+
+    So, our n-Gram overlap is:
+
+    $$
+    0.824^{0.25}\cdot0.625^{0.25}\cdot0.400^{0.25}\cdot0.214^{0.25} = 0.458
+    $$
+
+??? example "Step 3: Calculate Brevity Penalty"
+    We apply a brevity penalty to prevent the BLEU score from giving undeservingly high scores for short generated texts. Recall that our formula is:
+
+    $$
+    \min\left(1, \exp\left(1 - \frac{\text{reference length}}{\text{output length}}\right)\right)
+    = \min\left(1, \exp\left(1 - \frac{\text{23}}{\text{17}}\right)\right)
+    = 0.703
+    $$
+
+??? example "Step 4: Calculate BLEU"
+    Combining our n-Gram overlap and Brevity Penalty, our final BLEU score is:
+
+    $$
+    \text{BLEU} = \text{Brevity Penalty} \times \text{n-Gram Overlap} = 0.458 \times 0.703 = 0.322
+    $$
+
+    Note that in most cases, we may take the average of the BLEU score with respect to multiple reference texts - since multiple interpretations of the same sentences can be allowed. For example, if we calculated the BLEU score with the reference text being "Kolena is a comprehensive machine learning testing and debugging platform to surface hidden model behaviors and demystify model development", our BLEU score would be 0.571 - a much higher score from a small semantic change.
```

Comment:
Can we use something like code inline ` or code block ``` around the new reference text? I think it will make it more readable and stand out. (or maybe highlighting parts of the sentence where the semantic change is introduced.